zerver/actions/streams.py

```diff
@@ -159,21 +113,21 @@ def do_deactivate_stream(stream: Stream, *, acting_user: Optional[UserProfile])
     new_name = (hashed_stream_id + "!DEACTIVATED:" + old_name)[: Stream.MAX_NAME_LENGTH]
 
     stream.name = new_name[: Stream.MAX_NAME_LENGTH]
-    stream.save(update_fields=["name", "deactivated", "invite_only"])
+    stream.save(update_fields=["name", "deactivated"])
 
     assert stream.recipient_id is not None
     if was_web_public:
         assert was_public
         # Unset the is_web_public and is_realm_public cache on attachments,
-        # since the stream is now private.
+        # since the stream is now archived.
         Attachment.objects.filter(messages__recipient_id=stream.recipient_id).update(
             is_web_public=None, is_realm_public=None
         )
         ArchivedAttachment.objects.filter(messages__recipient_id=stream.recipient_id).update(
             is_web_public=None, is_realm_public=None
         )
```

Comment:
Should we remove this code because we don't mark streams as private anymore during deactivating or should the cache still be unset for archived streams?